### ISSUE TITLE
Error 404 this API does not exist

### ISSUE THREAD

**Issue by kcs32787github** [anxdpanic/plugin.video.twitch#537]
Getting errors trying to access the addon today. Havent changed anything with my setup. Error in Kodi says Failed to Connect to Twitch API. Nothing loads. Logs attached below.

Please provide any relevant information about your setup

* Add-on Version: 2.5.3
* Kodi Version: 18.8
* Kodi GUI Language: English
* Operating System: OSMC 2020.10-1
* Operating System Language: English

------

### Expected Behavior

See streams

------

### Current Behavior

Nothing loads at all

------

### Steps to Reproduce

Please provide detailed steps for reproducing the issue.

1. Open Addon

------

### Log

[https://paste.kodi.tv/gidocofaja.kodi](url)

------

### Additional Information

Worked as of 9:00 11/17/20 (GMT-5) Didn't work at 12pm (GMT-5) on 11/17/20

------

**Comment by kuraikaze** [comment]
Same error here - stopped working ~17:00GMT

Full debug log:
[https://paste.kodi.tv/idonerenop.kodi](url)

**Comment by neo-neo1** [comment]
Same error here on Kodi 18.9 on CoreELEC. Is there a fix? Thanks devs

**Comment by anxdpanic** [comment]
I've been using Twitch this whole time with no issues, and nothing conclusive in the logs or docs. 
It will likely pass, but keep me posted.

**Comment by kuraikaze** [comment]
For what it's worth - I'm in the UK, in case it's just part of the regional CDN having temporary issues.

**Comment by Agrajag-42** [comment]
If it helps, having the same issue in EU/Germany since today.

**Comment by kcs32787github** [comment]
I am US/Chicago Area. Just fyi.

**Comment by neo-neo1** [comment]
I’m also in the US with said issue

**Comment by pedro-santos** [comment]
EU/Portugal, also having this issue.
Twitch app and streamlink work fine though

**Comment by MrSprigster** [comment]
Happened to me also earlier this week.  Fixed it by removing the OAuth Token from Private API credentials.  It's likely the token expired, but I didn't get a new one so haven't confirmed, so could be an API change I suppose. My bets on expired token.

**Comment by Nigh7Stalk3r** [comment]
Had this same issue yesterday and it fixed itself, now in the last hour it's come back.

**Comment by danhofmann-com** [comment]
Here in US Florida, and having the same issue

Edit:

It happens on following live streams. I have a shortcut to that and it just fails… However, if I try to click on the plug-in itself under add-ons it also says connection failed.

**Comment by anxdpanic** [comment]
I can reproduce it only on Followed Games, the logs all look like Followed Live.

**Comment by MrSprigster** [comment]
I'm getting 404 error on the followed games. It was a different error on everything else until I cleared the private oauth. So probably unrelated to this issue. :).

**Comment by anxdpanic** [comment]
Just double checked and I don't have a private OAuth token entered. Will keep an eye on it.

**Comment by MrSprigster** [comment]
Yeah I think it was a separate issue. Disregard.

**Comment by Thrius** [comment]
Add-on Version: 2.5.3
This 404 error issue occurred yesterday, then app was okay until again today. Steps to reproduce:
1. 404 error when launching the twitch app with a previously generated oauth_token.
2. Cleared the token manually in the kodi/userdata/addon_data/plugin.video.twitch/settings.xml file
3. Twitch app now launches.
4. Generated a new oauth_token using url from settings.
5. 404 error when launching the twitch app

**Comment by kuraikaze** [comment]
Problem seems to have resolved it self for me as of ~23:00GMT
Live streams & VODs listing and playing normally.

**Comment by Nigh7Stalk3r** [comment]
Back working for me again now too.

**Comment by Thrius** [comment]
Same, issue resolved and able to login with updated oauth_token.

**Comment by neo-neo1** [comment]
Started working all on its own

**Comment by Nigh7Stalk3r** [comment]
This issue is back again in the last 30 minutes.
[kodi.log](https://github.com/anxdpanic/Twitch-on-Kodi/files/5561941/kodi.log)

**Comment by kcs32787github** [comment]
> This issue is back again in the last 30 minutes.
> [kodi.log](https://github.com/anxdpanic/Twitch-on-Kodi/files/5561941/kodi.log)

Likewise same here again. Seems like Twitch side of things, but I'm sure dev is keeping a close eye on things.

**Comment by neo-neo1** [comment]
This is the third consecutive day in a row that Twitch stops working with the same API error. For the past month Twitch (Amazon) has been aggressively cracking down hard on Adblockers and perhaps I’m paranoid but maybe they are targeting this Kodi Add-On as well since it doesn’t play ads properly. I’m sure Twitch (Amazon) has methods to detect Kodi. They have been doing some tricky stuff to counteract Adblockers as well. For all our sakes, I hope I’m wrong.

**Comment by gigalomaniaczero** [comment]
> This is the third consecutive day in a row that Twitch stops working with the same API error. For the past month Twitch (Amazon) has been aggressively cracking down hard on Adblockers and perhaps I’m paranoid but maybe they are targeting this Kodi Add-On as well since it doesn’t play ads properly. I’m sure Twitch (Amazon) has methods to detect Kodi. They have been doing some tricky stuff to counteract Adblockers as well. For all our sakes, I hope I’m wrong.

I was fearing this would happen too. Apparently, they are sending some code to disrupt the unofficial twitch clients. Hopefully, the devs will know how to deal with it.

**Comment by danhofmann-com** [comment]
Twitch needs to chill out… For some reason, this plug-in/API, won't even recognize my subscribed channels, so I get these "an ad is playing" or whatever it says all the darn time throughout streams that I am subbed to. And a preroll every single time I switch between streams, even if I just saw a pre-roll.

I'm sure the devs will be on it though… I wish I knew python better, or I would take a look as well… I'm also swamped with my day job :-(.

I use this plug-in all the time though, so hopefully twitch doesn't kill it

**Comment by Wibbbs** [comment]
Same issue here. Plugin version 2.5.3. Connection failed when loading the plugin. Same here I use this add on daily. Hopefully we don't lose it.

**Comment by gigalomaniaczero** [comment]
Hello,

Any news about this? Will there be any workarounds for this or the addon is killed?

**Comment by pezz** [comment]
I think given how the Twitch web site is working right now, where even if you just change to a different stream they insert an ad up your arse, I don't hold out much hope.

**Comment by danhofmann-com** [comment]
Streamlink (Python Twitch API streaming library and stream redirection CLI tool) still works I believe. They had issues last week with something similar to this, I believe… But it was resolved.

So since that is working still, I have no doubts that whatever is broken here can be fixed… just give the devs time. I am sure they have days jobs and whatnot.

I was actually thinking of working this and trying to integrate the stream link library into it and mapping the output of that API into the inputs for the rest of this plug-in, but I just don't have time… that would work in theory. And besides, I was thinking that would be a quick two hour fix to do… LOL, it's never that easy… Very appreciative of the current devs that work on this plug-in!

**Comment by neo-neo1** [comment]
I’ve been using this Kodi Add-On since it was released and I genuinely can’t remember the last time the Twitch API has went down for such a prolonged period. Here’s hoping, though knowing Twitch’s attack in Adblockers I surely won’t be holding my breath. 

Edit: Perhaps the dev can supply us with a new API key, when he has time.

**Comment by kcs32787github** [comment]
@anxdpanic could this information be relevant in any fashion- [Twitch Dev Post](https://discuss.dev.twitch.tv/t/future-for-followed-games-apis/29135) ? Seems to coincide (time-wise) with issue beginning with the addon. I know this points directly to the "Followed Games" feature, but perhaps it needs to be applied to the addon as a whole?

Edit: Potentially helpful info from streamlink as mentioned by @danhofmann-com: [Streamlink API overhaul](https://github.com/streamlink/streamlink/pull/3227/commits/ac7856c91fcbb6043a8ef8d160dbc135b51df179) . I have no idea if that is relevant or not, just trying to help. I don't code, but almost understand this stuff xD

Edit2: Also potential helpful info here from Streamlink- [Twitch API](https://github.com/streamlink/streamlink/pull/3344#issue-523498487)

**Comment by neo-neo1** [comment]
I can confirm the current API key is the problem as **I fixed it** (possibly temporarily) by using the FireFox Web Developer tools to extract the current `api_token` and `auth_token` from my logged in Twitch.tv account. I got the [idea from this post,](https://github.com/anxdpanic/Twitch-on-Kodi/issues/461#issuecomment-552035680) note that the posted API key does not work. I am not a dev sharing a fix, but this is how I currently got it to work. For all i know the API Key now changes daily, so I don't know how well of a fix it is. But, at least it currently does work.

Edit: More information on what I did is available in the first link inside this post. But here's more info anyway. If you want to manually edit the file, the [file location varies on which operating system and distribution](https://kodi.wiki/view/Userdata) you use. For me on CoreELEC it was in `/storage/.kodi/userdata/addon_data/plugin.video.twitch/settings.xml` (Hint: create a backup). The proper fields are `private_oauth_clientid` and `private_oauth_token` Note: read below for info on the probably optional`oauth_token`field.

If you don't want to manually edit the file, the proper fields in the Twitch Kodi Add-on are in Settongs -> Login -> `OAuth Client ID` and `OAuth Token` (the OAuth Token field at the bottom, not top). I believe the top `OAuth Token` field is optional, for subscribers VODs and etc... but I always create one just in case. Simply click on `Get OAuth Token` and follow the directions.

*Note: I found that preroll ads work as intended instead of the ad template that shows. Haven't watched long enough to test midroll ads.

**Comment by neo-neo1** [comment]
@anxdpanic I did create a new twitch dev project and tried inserting the respective API key, API Secret, and Oath token and the Add-on worked partially. I was able to successfully view my followed streamers but when I tried opening the stream the error I received was "invalid id client specified". Am I missing something? To be specific, I used the fields under "Login" and not "Developer". Can you please explain the purpose of the Developer fields vs Login fields?

**Comment by Wibbbs** [comment]
> I can confirm the current API key is the problem as I fixed it (possibly temporarily) by using the FireFox Web Developer tools to extract the current api_token and auth_token from my logged in Twitch.tv account. I got the [idea from this post,](https://github.com/anxdpanic/Twitch-on-Kodi/issues/461#issuecomment-552035680) note that the current posted API key does not work. I am not a dev sharing a fix, but this is how I currently got it to work. For all i know the API Key now changes daily, so I don't know how well of a fix it is. But, at least it currently does work.

I got my auth token and api token from a browser, where in the addon did you update these? Is there a settings config file somewhere?

**Comment by neo-neo1** [comment]
> > I can confirm the current API key is the problem as I fixed it (possibly temporarily) by using the FireFox Web Developer tools to extract the current api_token and auth_token from my logged in Twitch.tv account. I got the [idea from this post,](https://github.com/anxdpanic/Twitch-on-Kodi/issues/461#issuecomment-552035680) note that the current posted API key does not work. I am not a dev sharing a fix, but this is how I currently got it to work. For all i know the API Key now changes daily, so I don't know how well of a fix it is. But, at least it currently does work.
> 
> I got my auth token and api token from a browser, where in the addon did you update these? Is there a settings config file somewhere?

I edited my original post with answers to your question.

**Comment by Wibbbs** [comment]
So I was able to get this working by signing up for my own application on the twitch developer site and getting my own app client_ID and secret key. When I put these in the settings file, streams load now.

Issue I am having is I cant get a OAuth token since the token site seems to be using the redirectURI that I setup my for client_ID which I just set to localhost for now. Doesn't anyone know what the redirect should be?

In the mean time it seems to work fine without logged in functionality.

**Comment by neo-neo1** [comment]
> So I was able to get this working by signing up for my own application on the twitch developer site and getting my own app client_ID and secret key. When I put these in the settings file, streams load now.
> 
> Issue I am having is I cant get a OAuth token since the token site seems to be using the redirectURI that I setup my for client_ID which I just set to localhost for now. Doesn't anyone know what the redirect should be?
> 
> In the mean time it seems to work fine without logged in functionality.

That method doesn't open videos for me [(as I posted above)](https://github.com/anxdpanic/Twitch-on-Kodi/issues/537#issuecomment-730588762). What were the exact login fields that you populated inside settings.xml?

**Comment by Wibbbs** [comment]
```
<setting id="oauth_clientid">CLIENTID_GOES_HERE</setting>
    
<setting id="oauth_token" default="true">CLIENTSECRET_GOES_HERE</setting>
```

However I just noticed that while the client_id it still present in my setting file, the oauth_token seems to have been removed and is empty now.

`<setting id="oauth_token" default="true"></setting>`

**Comment by neo-neo1** [comment]
> ```
> <setting id="oauth_clientid">CLIENTID_GOES_HERE</setting>
>     
> <setting id="oauth_token" default="true">CLIENTSECRET_GOES_HERE</setting>
> ```
> 
> However I just noticed that while the client_id it still present in my setting file, the oauth_token seems to have been removed and is empty now.
> 
> `<setting id="oauth_token" default="true"></setting>`

Yep that method doesn't work for me. I'll just use the other method for now

**Comment by danhofmann-com** [comment]
@neo-neo1 

I am attempting to use your work around, but putting the api_key (from dev tools) into the private_oauth_clientid field and the auth-token (from dev tools) into the private_oauth_token field (and leaving oauth_clientid and oauth_token blank) does not work.  I get Authentication Failed, when i do this.

If I put the api_key into oauth_clientid, i get "invalid client id specified"

What did you specify for the 4 fields below?

    <setting id="oauth_clientid" default="true"></setting>
    <setting id="oauth_token"></setting>
    <setting id="private_oauth_clientid"></setting>
    <setting id="private_oauth_token"></setting>


Edit: By putting the cookie values into the fields above, the plugin works (whereas I can view Featured Streams), but I cannot view Followed Streams, as it is not actually logging me in. I have a favorite on the main Kodi screen for Live Channels, and when I click on that, I get a auth failure...

**Comment by danhofmann-com** [comment]
I am assuming this is a similar issue...

https://github.com/streamlink/streamlink/issues/3343

**Comment by gigalomaniaczero** [comment]
If I get new OAuth token and ClientID, will it fix the problem?

**Comment by kcs32787github** [comment]
@anxdpanic Any status update on this? People are posting their own troubleshooting/ideas, but none of which seem to result in a fully functioning addon. 
I personally would like to wait and hear from you about either a potential workaround, or an update coming to the addon to patch things up.
Also, if there is anyway I can support the development of this addon financially, please let me know how to do that.
Thanks!

**Comment by neo-neo1** [comment]
> @anxdpanic Any status update on this? People are posting their own troubleshooting/ideas, but none of which seem to result in a fully functioning addon.

Works for me and a few others that I’ve told. It seems the current issue is not a problem with the Add-On but an old setting (API) not working anymore which needs to be changed by the end user. But I do agree, an official statement is best.

**Comment by gigalomaniaczero** [comment]
Do I need to create a new developer project in order to get client ID? Previously, I had client ID in the addon without me having to create anything manually.

**Comment by scottismyname** [comment]
@neo-neo1 
Sorry for the trouble but can you explain exactly what fields in the settings come from where? I have both fields extracted from Firefox but not sure what settings should be filled in with what.

In the settings menu there are many options:
Under general there are the top oauth token, and then at the bottom there is oauth client id and another oauth token.
Under developer, there is another oauth client id

Can you clarify which fields are blank and what are filled in?

**Comment by anxdpanic** [comment]
@neo-neo1 
The OAuth Token is your generated oauth token
The private credentials are credentials from the web client, private client id must be `kimne78kx3ncx6brgo4mv6wki5h1ko`. 
The private oauth token is the `auth-token` from the websites cookies and required for subscribers/turbo perks.

The developer section is used when replacing the oauth process including the redirect uri. This is unlikely what anyone wants to use.

@kcs32787github and others
As for this issue, the only end point that doesn't work for me is `Following Games` (and it's not a public api and expected to break).  Playback is also working, I get a commercial countdown instead of the commercial but they are only shown when I would expect them to be shown. All of this has been working the whole time for me.

So it seems like either a Twitch change hasn't completely propagated, or users settings have been messed up try to fix an intermittent issue.

Anyone that wants to fix their settings after all this messing around;
Check that `Settings - Login -  OAuth Token` is your generated OAuth token if you have generated one (optional).
Check that `Settings - Login - Private API Credentials - OAuth Client ID` is `kimne78kx3ncx6brgo4mv6wki5h1ko`  (required)
Check that `Settings - Login - Private API Credentials - OAuth Token` is left blank, or  the `auth-token` cookie from your web browser when logged into Twitch

All settings under  `Settings - Developers` should be left blank. (required)

**Comment by gigalomaniaczero** [comment]
I have tried everything and nothing works. I always get "failed to connect to Twitch API" :(

**Comment by Cnuut** [comment]
I reinstalled the twitchaddon
also deleted the twitch setting.xml ("rm .kodi/userdata/addon_data/plugin.video.twitch/settings.xml")
and did "Get OAuth token" open link in a browser, then you get the 'token', put that in the top 'OAuth Token' field, press OK
now it works again! (for me)

**Comment by neo-neo1** [comment]
To be clear, the old clientid `kimne78kx3ncx6brgo4mv6wki5h1ko`
does not work for me (it was the settings I had when the API errors started) and it’s what I had to replace from my own Twitch.tv logged in user account with FireFox Web Tools. As well as the private token. Once that’s replaced as well as a new token is generated from Twitch Add-On settings everything works as expected. Just in case,  I also reset the function cache from Twitch Add-On settings was when I changed keys. 

Edit: For those asking for clarification I’m  sorry but I don’t know how to be more clearer, I already included the specific fields I changed in `settings.xml` in a previous post for such a reason. In the Twitch Add-On settings they translate to both the fields under `Private API credentials`. Once both fields are manually filled in, click on `Get OAuth Token` which is for the top `OAuth Token` field. 

Try clearing all your fields, resetting your function cache (under maintenance in Twitch Add-on settings), and even perhaps restart Kodi. To be safe you can even clear all caches listed under maintenance in the Twitch Add-On settings. Deleting your `settings.xml` file is a bit barbaric and redundant. If you get an error when clicking `Get OAuth Token` that is a symptom of having improper API settings. 

If the Twitch Add-On starts to work but you can’t open any followed streams for me that occurred when I improperly  inserted the API keys, most likely the top OAuth Token.

**Comment by kcs32787github** [comment]
@anxdpanic After your suggestion, I found that my oath_token had seemingly expired. I followed @Cnuut suggestion and deleted my user_addon settings.xml file. Then uninstalled the addon. Next I rebooted my device. I reinstalled the addon from scratch. Only then upon clicking Get Oath Token from the settings page could I get the link information to appear within Kodi. I obtained a new Oath Token from the link and my web browser. Input that into the addon and everything is working again. 

Before deleting the settings.xml file and clicking on Get Oath Token resulted in seeing the same Error 404 msg appearing within  Kodi. So that step is crucial.

**Comment by kcs32787github** [comment]
So long story short, people experiencing this specific issue, you likely need to regenerate a new Oath Token and replace the one at the very top of the settings 'Login' page within the addon.

**Comment by gigalomaniaczero** [comment]
I have copied my Oauth from my Twitch account and it doesn't work. I don't know what am I doing wrong. Where do I get a new Client ID that works if the kimne78kx3ncx6brgo4mv6wki5h1ko has expired?

**Comment by anxdpanic** [comment]
@neo-neo1 and @gigalomaniaczero 
The `kimne78kx3ncx6brgo4mv6wki5h1ko` key affects playback only and isn't used for listing your following stuff and seems to be working absolutely fine. @Cnuut and @kcs32787github replies confirm that as well.

**Comment by Lucki** [comment]
> I found that my oath_token had seemingly expired.

This is the solution, you have to create a new oauth token.
Sadly the link in the settings doesn't work with the same api error message. I've created the link with a different kodi installation.

**Comment by Cnuut** [comment]
The OAuth link didnt work for me at first, after I reinstalled the twitchapp it did (same exact link) very strange.

**Comment by gigalomaniaczero** [comment]
Ok, I finally fixed it. Everything is working ok now. Thanks everyone, much appreciated.

**Comment by danhofmann-com** [comment]
OK, I can confirm, Expiring OAuth (the one you get from Twitch from clicking "Get OAuth Token", was the entire problem (and before when I clicked on it, the Plugin gave me a 404, so I just uninstalled/reinstalled the plugin, and presto).

You do have to use "kimne78kx3ncx6brgo4mv6wki5h1ko" as the private oauth client id.

I did what others did. I completely removed the plugin and settings, and then reinstalled the plugin.
Upon doing this, the oauth client id of "kimne78kx3ncx6brgo4mv6wki5h1ko" is automatically put in the private oauth client field. You don't need to do anything.

Next, click Get OAuth Token link, follow the steps in your web browser, obtain the token, and paste it in (for some reason I couldn't add it to the GUI settings screen, so I just added it to my settings file).

That's ALL you need. Once you do that, EVERYTHING works again. Commercial breaks show the "Add is Playing" screen, but not an actual ad.

Now before this all went to hell, I had actually pasted in my auth-token from the Twitch Website cookie into the private oauth token field. Leaving this blank, works, and you just get that blue-ish purple-ish ad loading screen.

However, what is happening now, is if you DO copy your auth-token from the Twitch Website cookie into the private oauth token field, it DOES PLAY REAL ADS. 

Very strange. I don't know if leaving your private oauth token field (again, this is the bottom field), blank, versus adding in your auth-token, has any other behaviors, other than actually playing ads, instead of showing the ad loading screen.

If anyone knows if its better to add your auth-token, or just leave the private oauth token field blank, let me know.

Either way, the OAuth Token expiring appeared to be the whole issue this whole time, for me, and others (I guess Twitch expired a lot of them, maybe for US based users?). Plugin is now working again. Thanks to everyone on this thread.

**Comment by dpetka2001** [comment]
Even after doing all that was suggested in this ticket, the following games endpoint still doesn't work? But as I read above it's due not being public API and expected to break. Everything else works really good. Should we expect a fix for following games or is there nothing to be done?

**Comment by xenius** [comment]
Same, not working with reinstall for me.

**Comment by Wibbbs** [comment]
New Oauth token worked for me. I removed the add on and re installed. Got a new oauth token and everything seems to be working properly. Thanks all! 👍🏼

**Comment by mhorst314** [comment]
> OK, I can confirm, Expiring OAuth (the one you get from Twitch from clicking "Get OAuth Token", was the entire problem (and before when I clicked on it, the Plugin gave me a 404, so I just uninstalled/reinstalled the plugin, and presto).
> 
> You do have to use "kimne78kx3ncx6brgo4mv6wki5h1ko" as the private oauth client id.
> 
> I did what others did. I completely removed the plugin and settings, and then reinstalled the plugin.
> Upon doing this, the oauth client id of "kimne78kx3ncx6brgo4mv6wki5h1ko" is automatically put in the private oauth client field. You don't need to do anything.
> 
> Next, click Get OAuth Token link, follow the steps in your web browser, obtain the token, and paste it in (for some reason I couldn't add it to the GUI settings screen, so I just added it to my settings file).
> 
> That's ALL you need. Once you do that, EVERYTHING works again. Commercial breaks show the "Add is Playing" screen, but not an actual ad.
> 
> Now before this all went to hell, I had actually pasted in my auth-token from the Twitch Website cookie into the private oauth token field. Leaving this blank, works, and you just get that blue-ish purple-ish ad loading screen.
> 
> However, what is happening now, is if you DO copy your auth-token from the Twitch Website cookie into the private oauth token field, it DOES PLAY REAL ADS.
> 
> Very strange. I don't know if leaving your private oauth token field (again, this is the bottom field), blank, versus adding in your auth-token, has any other behaviors, other than actually playing ads, instead of showing the ad loading screen.
> 
> If anyone knows if its better to add your auth-token, or just leave the private oauth token field blank, let me know.
> 
> Either way, the OAuth Token expiring appeared to be the whole issue this whole time, for me, and others (I guess Twitch expired a lot of them, maybe for US based users?). Plugin is now working again. Thanks to everyone on this thread.

This worked for me too. One note on adding your auth-token: I was not able to access subscriber-only content without a private oauth token, so I did not leave that field blank.

Also, I have multiple systems with the twitch plugin. Once one system was up and working again, simply copying the settings.xml   from the working system to the other systems got them working again. So there was no need to re-install or run to the process again for the other systems.

**Comment by xenius** [comment]
Reinstalled addon, removed plugin settings, created new token... Nothing is working for me. when i click to show my followed games i get this error... Can anyone please tell me what i can do to fix this? What people suggesting in this post is not working for me.

**Comment by anxdpanic** [comment]
@xenius `followed games` won't work afiak, everything else should

**Comment by xenius** [comment]
Okay! Thanks for clarifying this.

**Comment by IdotMaster1** [comment]
where do i find the settings file? I really want to get twitch working on my iPad 2 running iOS 6

**Comment by anxdpanic** [comment]
Following games should be fixed in the latest beta for users with private oauth tokens.

https://github.com/anxdpanic/Twitch-on-Kodi/releases/tag/2.5.4-dev
https://github.com/anxdpanic/Twitch-on-Kodi/wiki/Private-API-Credentials---OAuth-Token

**Comment by dpetka2001** [comment]
It works perfectly. Thank you very much for your work :smile:

**Comment by anxdpanic** [comment]
https://github.com/anxdpanic/script.module.python.twitch/releases/tag/2.0.16-dev

**Comment by dpetka2001** [comment]
Can we just update twitch on kodi and it will also install the latest script.module.python.twitch? Or how should we install it? Just download and replace the old one in addons folder?

**Comment by anxdpanic** [comment]
For the moment, download the zip and install from zip.
https://kodi.wiki/view/Add-on_manager#How_to_install_from_a_ZIP_file

**Comment by kcs32787github** [comment]
> https://github.com/anxdpanic/script.module.python.twitch/releases/tag/2.0.16-dev

I started getting api errors this afternoon. I installed this update and the issue was resolved. Thanks!

**Comment by hiemsmutus** [comment]
Which zip file do I use, script.module.python.twitch-2.0.16+matrix.1.beta2.zip or script.module.python.twitch-2.0.16.beta2.zip?
Do you have a repo I can add?

**Comment by anxdpanic** [comment]
the `+matrix` version is for Kodi 19 (Matrix), the other is for any other version of Kodi

The twitch development repo is available from https://panicked.xyz/repositories/ the `matrix` folder for Kodi 19

**Comment by M-Reimer** [comment]
Any solution to this? I'm on the latest version available from the Kodi repository and since yesterday I can no longer play any streams.

**Comment by MrSprigster** [comment]
The solution is installing the following (using install from zip in Kodi interface):
https://github.com/anxdpanic/script.module.python.twitch/releases/tag/2.0.16-dev

For Kodi 19 Matrix: https://github.com/anxdpanic/script.module.python.twitch/releases/download/2.0.16-dev/script.module.python.twitch-2.0.16+matrix.1.beta2.zip

For previous versions of Kodi: 
https://github.com/anxdpanic/script.module.python.twitch/releases/download/2.0.16-dev/script.module.python.twitch-2.0.16.beta2.zip

**Comment by dpetka2001** [comment]
I'm also on the latest twitch for kodi 2.5.4-dev version. In the default kodi repos the version is 2.5.3. So maybe that is the problem with some ppl? After having installed twitch 2.5.4-dev version (which you can download from the releases on this repo) as a zip and also latest python twitch module from the links mentioned above also as a zip file, everything works like a charm without any problems at the moment.

**Comment by M-Reimer** [comment]
Is there any plan when we will see a fixed version from the official repos? Currently the officially shipped "stable" version is broken...

**Comment by elgatito** [comment]
Installed this: https://github.com/anxdpanic/script.module.python.twitch/releases/tag/2.0.16-dev
and this: https://github.com/anxdpanic/Twitch-on-Kodi/releases/tag/2.5.4-dev

Updated private oauth token (to be sure), and still getting 404 error for video playbacks.

**Comment by MrSprigster** [comment]
@elgatito Make sure you have installed beta2 of python twitch (see my comment above for the zip link). Let us know how you get on.

**Comment by MrSprigster** [comment]
@anxdpanic Do you think the API changes have propagated far enough for the fix to be generally released?

**Comment by M-Reimer** [comment]
Installed the mentioned development releases and now Twitch works for me again :+1:

**Comment by elgatito** [comment]
@MrSprigster Ouch, thought beta2 is for "old" Kodi.
With beta2 everything works!

**Comment by MrSprigster** [comment]
Glad to hear it :) 👍

**Comment by Fumi24** [comment]
The install from zip menu is empty, I am on OSMC November 2020 2020.11 version, and I think I am running kodi 18.9, dunno what I am doing wrong. Using the https://github.com/anxdpanic/script.module.python.twitch/releases/tag/2.0.16-dev link

Edit: the beta2.zip link works but cant make it install the plugin.

**Comment by MrSprigster** [comment]
👍

**Comment by MrSprigster** [comment]
@Fumi24  use install from zip to install the latest dev version of Twitch on Kodi (link below):

https://github.com/anxdpanic/Twitch-on-Kodi/releases/download/2.5.4-dev/plugin.video.twitch-2.5.4.beta1.zip

And for anyone on Matrix ( https://github.com/anxdpanic/Twitch-on-Kodi/releases/download/2.5.4-dev/plugin.video.twitch-2.5.4+matrix.1.beta1.zip )

To clarify, you need this dev version of Twitch on Kodi aswell as the Beta2 of Python Twitch (links to that in my previous post ^)

**Comment by Fumi24** [comment]
@MrSprigster I think OSMC is doing some weird stuff, when I put the beta1.zip in my source it shows up as a empty file. I am not the best at doing things of SSH or any Pi related stuff so I think I will just wait for a fix to be fully deployed :) thanks for everything on the addon its great! used it for many years.

Fixed it using wget on a SSH connection 👍

**Comment by MrSprigster** [comment]
Great News 👍

**Comment by joshsmithtogo** [comment]
Those links for beta releases are 404 now.  What's up with that?

**Comment by anxdpanic** [comment]
I knew I'd forget something hah, https://github.com/anxdpanic/Twitch-on-Kodi/releases/tag/2.5.4

**Comment by joshsmithtogo** [comment]
> I knew I'd forget something hah, https://github.com/anxdpanic/Twitch-on-Kodi/releases/tag/2.5.4

Thanks anxdpanic.  That did the trick.

**Comment by sebastianEriksson87** [comment]
Now its working agian!
@anxdpanic u r the best, thx!

**Comment by anxdpanic** [comment]
The updates will be available in ~8 hrs from the official repository.

**Comment by cwildfoerster** [comment]
Error still persists with 2.5.4 / 2.0.16 @anxdpanic 

```
2021-01-21 16:54:38.643 T:1937092624  NOTICE: ADDON: plugin.video.twitch v2.5.4 installed
2021-01-21 16:54:38.646 T:1937092624  NOTICE: ADDON: script.module.python.twitch v2.0.16 installed
....
2021-01-21 17:01:33.384 T:1453663104 WARNING: script.module.python.twitch: API version |V5| is deprecated, update to |Helix| by |TBD|
2021-01-21 17:01:36.026 T:1453663104   ERROR: Twitch: Error |Not Found| |[404] This API does not exist|

```

**Comment by fayzeshyfft** [comment]
Just want to add something I did that I don't think has been mentioed yet.
I was running into all of the issues in this thread and decided to try it fresh from the start. Cleand the Twitch addon completely from Kodi. Without adding an Auth token it worked but when using the auth token from the cookies it still would fail saying that it could not connect to the Twith API.
So this time I went into the settings of my Twitch account and went to Connections > Other Connections. In there was TTV on Kodi and I clicked on Disconnect. After doing that I went into the Kodi settings and used the URL you get from the Get OAuth token option in there and re-authorized TTV on Kodi. It gave me an Auth token there and upon entering that into the addon settingss everything worked again!
